examples/3d_drawing/3d_rotation_anime.py

```diff
@@ -0,0 +1,45 @@
+"""
+===========================
+3D Rotating Graph Animation
+===========================
+
+3D spring plot.
+
+Animation is adapted from
+https://matplotlib.org/gallery/animation/basic_example.html
+https://sphinx-gallery.github.io/stable/auto_examples/plot_8_animations.html
+
+"""
+import numpy as np
+import matplotlib.pyplot as plt
+from matplotlib import animation
+
+###############################################################################
+# Create a sample plot.
+# ---------------------
+#
+# Temp plot for testing doc generation from
+# https://github.com/networkx/networkx/blob/main/examples/drawing/plot_simple_path.py
+
+G = nx.path_graph(8)
+pos = nx.spring_layout(G, seed=47)  # Seed layout for reproducibility
+nx.draw(G, pos=pos)
+plt.show()
+
+###############################################################################
+# Create a sample animation.
+# --------------------------
+#
+# Test animation plot for testing doc generation for animation.
+
+
+def _update_line(num):
+    line.set_data(data[..., :num])
+    return (line,)
+
+
+fig, ax = plt.subplots()
+data = np.random.RandomState(0).rand(2, 25)
+(line,) = ax.plot([], [], "r-")
+ax.set(xlim=(0, 1), ylim=(0, 1))
+ani = animation.FuncAnimation(fig, _update_line, 25, interval=100, blit=True)
```

Comment:
Does this need a `plt.show()` at the end?
No image is showing the in gallery.

Reply:
in the original
https://sphinx-gallery.github.io/stable/auto_examples/plot_8_animations.html
it does not

Reply:
But currently it does not even show
```
G = nx.path_graph(8)
pos = nx.spring_layout(G, seed=47)  # Seed layout for reproducibility
nx.draw(G, pos=pos)
plt.show()
```
copied from ttps://github.com/networkx/networkx/blob/main/examples/drawing/plot_simple_path.py

Reply:
I think it shows the plot in [the gallery examples.](https://output.circle-artifacts.com/output/job/6c3f190f-df9b-4434-889d-d917f52b01d2/artifacts/0/doc/build/html/auto_examples/drawing/plot_simple_path.html#sphx-glr-auto-examples-drawing-plot-simple-path-py)

Are you looking at the "documentation artifact" generated by a circle-ci action "check"? Clicking "Details" on the  report of the passing test labeled artifact brings you to the fully rendered documentation. You probably know that already. Am I missing something, because the simple path seems to be produced.

The example you show of a sphinx generated matplotlib animation has much more than an animated gif file. It has a bunch of control buttons for the viewer to step through the animation, etc. 

The example ["matplotlib unchained"](https://matplotlib.org/stable/gallery/animation/unchained.html) also uses `plt.show()`.  But those don't use the `ani` toolset.  I'm not sure what the difference between these examples is. And the examples don't show the sphinx side of the setup.

Reply:
You refer to the previously built example which may be kept untouched since there is no change in this old module. I refer to https://output.circle-artifacts.com/output/job/6c3f190f-df9b-4434-889d-d917f52b01d2/artifacts/0/doc/build/html/auto_examples/3d_drawing/3d_rotation_anime.html that reproduces this same example as a first plot in my new module. There are no plots here and "Total running time of the script: (0 minutes 0.000 seconds)" so probably the module never runs.

Reply:
The same story in https://github.com/networkx/networkx/pull/7040

Reply:
@dschult After renaming the module to plot_ I got the first animation, from 
https://sphinx-gallery.github.io/stable/auto_examples/plot_8_animations.html
to run in my module and produce the plot just like in the reference with all the control buttons. There is `plt.show()` at the end unless you need to run it locally,

My actual animations have some mistakes not visible locally - trying to fix them now. But generally, animation now works in this PR, just need to fix all bugs.

Reply:
Yay!  Great progress!